src/Hintman/HLint.hs

```diff
@@ -0,0 +1,91 @@
+{- | This module introduces function to work with @HLint@ tool at modified files
+in PRs.
+-}
+
+module Hintman.HLint
+       ( runHLint
+       ) where
+
+import Language.Haskell.HLint4 (Idea, applyHints, autoSettings, parseModuleEx)
+import Network.HTTP.Client (Response (..), httpLbs)
+import Network.HTTP.Client.TLS (newTlsManager)
+import Network.HTTP.Types (Status (..))
+import System.FilePath (takeExtension)
+import Text.Diff.Parse.Types (FileDelta (..), FileDeltas, FileStatus (..))
+
+import Hintman.Core.PrInfo (Branch (..), Owner (..), PrInfo (..), Repo (..))
+
+import qualified Data.Text as T
+
+
+{- | Run @HLint@ on all modified files.
+
+For this we need:
+
+1. Filter all modified files.
+2. Filter only `*.hs` extension files.
+3. Get the content of all modified files.
+4. Run @Hlint@ tool on the content.
+5. Return the list of all 'Idea's.
+
+-}
+runHLint :: MonadIO m => PrInfo -> m [Idea]
+runHLint prInfo@PrInfo{..} = do
+    let modFiles = filter ((==) ".hs" . takeExtension) $ getModifiedFiles prInfoDelta
+    let getContent = downloadFile . createFileDownloadUrl prInfo
+    fileContent <- traverse (\x -> getContent x >>= \y -> pure (x, y)) modFiles
```

Comment:
Here you can use `traverseToSnd` from `relude`.

* https://hackage.haskell.org/package/relude-0.5.0/docs/Relude-Extra-Tuple.html#v:traverseToSnd

so the code will look like:

```haskell
traverse (traverseToSnd getContent) modFiles
```

But I'm not sure how readable it is... Can be just `(x,) <$> getContent x`